jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/persistence/QueryDefinitionEntity.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.kie.services.impl.query.persistence;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Lob;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+
+import org.jbpm.kie.services.impl.query.SqlQueryDefinition;
+import org.jbpm.services.api.query.model.QueryDefinition;
+import org.jbpm.services.api.query.model.QueryDefinition.Target;
+
+@NamedQueries(value={
+        @NamedQuery(name="getQueryDefinitions", query="select qd from QueryDefinitionEntity qd"),
+        @NamedQuery(name="getQueryDefinitionByName", query="select qd from QueryDefinitionEntity qd where name = :name")
+})
+@Entity
```

Comment:
I've just noticed that this is the 3rd use of `@NamedQueries` that we have: 
- DeploymentStoreEntry
- ContextMappingInfo

For maintainability and readability, I would much rather have **all** of our queries either be via annotation **or* in xml. Again, if we were a small project, this wouldn't matter. But we're not -- there are thousands and thousands of LoC. 

IMHO, choosing one way to do things (xml via anno's) is part of saving energy so that we can spend it on the important issues. 

Please move these to an xml config -- or, if you think we should do everything via anno's, then that's also fine, but then we need to do that as well.